starter-code/src/clue.js

```diff
@@ -1,94 +1,172 @@
-ocupation:   Scientist
+  
+  // Characters Collection
+  var charactersArray = [mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard];
+  
+  // Rooms' Collection
+  var roomsArray = [diningRoom, conservatory, kitchen,  study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio];
+  
+  // Weapons Collection
+  var weaponsArray = [rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol];
+  
+  function randomSelector(arr){
+    var random = arr[Math.floor(Math.random() * arr.length)];
+    return random;
+  }
 
-profPlum
-first_name:   Victor
-last_name:    Plum
-color:        purple
-description:  Billionare video game designer
-age:          22
-image:        https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg
-occupation:   Designer
+  function pickMistery() {
+    var mysteryEnvelope = [];
+    mysteryEnvelope.push(randomSelector(charactersArray));
+    mysteryEnvelope.push(randomSelector(weaponsArray));
+    mysteryEnvelope.push(randomSelector(roomsArray));
+    return mysteryEnvelope;
+  }
+  
+  var mysteryEnvelope = pickMystery();
 
-missScarlet
-first_name:   Kasandra
-last_name:    Scarlet
-color:        red
-description:  She is an A-list movie star with a dark past
-age:          31
-image:        https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg
-occupation:   Actor
-
-mrsPeacock
-first_name:   Eleanor
-last_name:    Peacock
-color:        blue
-description:  She is from a wealthy family and uses her status and money to earn popularity
-age:          36
-image:        https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg
-occupation:   Socialité
-
-mrMustard
-first_name:   Jack
-last_name:    Mustard
-color:        yellow
-description:  He is a former football player who tries to get by on his former glory
-age:          62
-image:        https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg
-occupation:   Retired Football player
-
-// Weapons
-
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
-
-// Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
-
-// Characters Collection
-var charactersArray = [];
-
-// Rooms' Collection
-var roomsArray = [];
-
-// Weapons Collection
-var weaponsArray = [];
+  function revealMistery(arr) {
+    var string = ""
+    string += arr[0].first_name + " " + arr[0].last_name + " killed Mr.Boddy using the " + arr[1].name + " in the " + arr[2].name + "!!!!";
+    return string
+  }
```

Comment:
You're missing something to execute code:
```
console.log(revealMystery(pickMystery(charactersArray, weaponsArray, roomsArray)));
```

Reply:
oh yeah, it passes the jasmine text but wouldn't work on its own. my bad, was being myopic.